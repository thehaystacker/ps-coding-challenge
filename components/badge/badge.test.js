import React from "react";
import "@testing-library/jest-dom";
import { render, within } from "@testing-library/react";
import Badge from "./index";

describe("Badge", () => {
  test("should render the component without crashing", () => {
    const { getByTestId } = render(<Badge></Badge>);
    const message = getByTestId("badge");

    expect(message).toBeVisible();
  });

  test("should render the badge text", () => {
    const { getByTestId } = render(<Badge>2016</Badge>);
    const { getByText } = within(getByTestId("badge"));
    const message = getByText(`2016`);

    expect(message).toBeInTheDocument();
  });

  test("should match the snapshot", () => {
    const { asFragment } = render(<Badge>2016</Badge>);
    expect(asFragment()).toMatchSnapshot();
  });
});
