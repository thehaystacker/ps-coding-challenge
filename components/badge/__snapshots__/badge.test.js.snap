// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should match the snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="badge"
  >
    <span>
      2016
    </span>
  </div>
</DocumentFragment>
`;
